File: Backend/Models/WatchListMovie.cs
namespace Movie_WatchList.Models;

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class WatchlistMovie
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.None)] // Disable auto-increment
    public int Id { get; set; }  // Primary Key

    [Required]
    [MaxLength(255)]
    public string Title { get; set; }

    [Required]
    public string Overview { get; set; }

    public string PosterPath { get; set; }

    [Column(TypeName = "decimal(3,1)")]
    public double Rating { get; set; }

    // Watchlist-specific fields
    public bool Watched { get; set; }
    public DateTime AddedDate { get; set; }
}

File: Frontend/src/models/watch-list-movie.ts
export class WatchlistMovie {
    id: number; // Primary Key
    title: string;
    overview: string;
    posterPath?: string;
    rating: number;
    watched: boolean;
    addedDate: Date;
  
    constructor(
      id: number,
      title: string,
      overview: string,
      rating: number,
      watched: boolean,
      addedDate: Date,
      posterPath?: string
    ) {
      this.id = id;
      this.title = title;
      this.overview = overview;
      this.posterPath = posterPath;
      this.rating = rating;
      this.watched = watched;
      this.addedDate = addedDate;
    }
  }
